tools/rp_storage_tool/src/bucket_reader.rs

```diff
@@ -1873,13 +1922,22 @@ impl BucketReader {
                 },
                 revision_id: partition_revision,
             };
+            let labeled_ntpr = LabeledNTPR {
+                ntpr,
+                // Prefixes of exactly 8 characters are hashes rather than labels.
+                label: if prefix.len() != 8 {
+                    Some(prefix)
+                } else {
+                    None
+                },
+            };
```

Comment:
nit: since we have this pattern in multiple places, it might be worth it to have an impl method to create a labelled ntpr given a prefix?

Reply:
Done